iam.tf

```diff
@@ -15,9 +15,11 @@ resource "aws_iam_role" "ptfe" {
   ]
 }
 EOF
+
```

Comment:
do we as a team have existing thoughts on using the data:aws_iam_policy_document for things like this over HEREDOC syntax.

I have my own thoughts (that keeping as much HCL in an HCL file is better than trying to switch to json) but I'd go with the prevailing winds

for example the above would be 
```
data "aws_iam_policy_document" "assume_role" {
  statement {
    actions = ["sts:AssumeRole"]
    effect = "Allow"

    principals {
      type = "Service"
      identifiers = ["ec2.amazonaws.com"]
    }
  }
}


resource "aws_iam_role" "ptfe" {
  name = "ptfe-${module.common.install_id}"
  name = "ptfe-${module.common.install_id}"
  assume_role_policy = data.aws_iam_policy_document.assume_role.json
}
```

Reply:
I like keeping as much in HCL as possible rather than doing messy interpolation. Ended up bringing over the other branch changes that went ahead and made this chance.